CONTRIBUTING.md

```diff
@@ -2,31 +2,112 @@
+quick [search][2] - you may not be alone! If you aren't having any luck, then go ahead and open a new issue by selecting the
+"Feature" issue type and answer some needed questions.
 
 ## Code review process
 
-The core team looks to Pull Requests on a regular basis and in a best effort.
+The core team looks at Pull Requests on a regular basis on a best-effort basis.
+
+## How are Github issues handled
+
+### Queues
+
+1. icebox: contain new issues along with issues that are waiting for grooming and not read to start working yet. Some issues
+may be closed or rejected.
+
+2. todo: issues ordered by priority and ready to be worked on by contributors.
+
+3. in progress: issues being implemented.
+
+4. under review: issues waiting to be reviewed.
+
+5. done: all issues closed.
+
+### Labels
+
+The labels are grouping in: **priority**, **status** and **type**. There are some individual labels like
+**good first issues** or **help wanted** that can be used but are not tracked from a project management perspecitve.
+
+All the grouping labels must contain a short description to help contributors to understand better the intention of
+each one.
+
+#### Status
+
+1. **Accepted**: issue was accepted and it may be implemented in near future.
+2. **Need More Info**: issue requires more information to be evaluated.
+3. **Verification Needed**: issue probably waiting for a PR to be merged.
+4. **Validation**: issue has enough information but it requires a discussion before accepted.
+
+[todo] more about the labels grouping in future versions of the document.
+
+
+### Creating New Issues
+
+The issue created on Github will be automatically assigned to the product owner, who will then evaluate the issue from an
+information-gathering perspective.
+
+If an issue doesn't contain any description and/or if the same is not enough to assess, then more information will be
+requested to the issue author.
+
+In case of an issue gets labeled "Stale" (more than 60 days without any activity), the product owner will:
+
+1. close the issue in case of lack of information.
+2. close the issue if it's no longer relevant.
+3. change labels if it's still relevant and information is sufficient.
+
+### Where to start
+
+All contributions are welcome but the best place to pick an issue to work on is from the _todo_ queue that is ordered by
+priority. Before you move it to the _in progess_ queue, make sure that the story is clear. If not, drop comments on the
+issue.
+
+### The flow
+
+The issue flow is always to the right that means if you pick an issue from the _todo_ queue, you should work on it until
+it's ready to be reviewed by a peer.
+
+If you need to move the issue back to a previous queue, then please leave a clear comment stating the reasons so the
+project core contributors can refine it.
+
+1. pick an issue from the _todo_ queue and move it to the in progress.
+2. add your name to the assign field and other members name if they are pairing with you.
```

Comment:
s/assign/assignee ?

Reply:
`member's names` (possessive `'s`, and plural `s`)